dask_expr/io/parquet.py

```diff
@@ -693,6 +700,123 @@ def fs(self):
                 fs = type(fs)(**region, **storage_options)
             return fs
 
+    def approx_statistics(self) -> dict:
+        """Return an approximation of a single files statistics.
+
+        This is determined by sampling a few files and averaging their statistics.
+
+        Fields
+        ------
+        num_rows: avg
+        num_row_groups: avg
+        serialized_size: avg
+        columns: list
+            A list of all colum statistics where individual fields are also
+            averaged.
+
+
+        Example
+        -------
+        {
+            'num_rows': 1991129,
+            'num_row_groups': 2.3333333333333335,
+            'serialized_size': 6256.666666666667,
+            'total_byte_size': 118030095,
+            'columns': [
+                {'total_compressed_size': 6284162.333333333,
+                'total_uncompressed_size': 6347380.333333333,
+                'path_in_schema': 'l_orderkey'},
+                {'total_compressed_size': 9423516.333333334,
+                'total_uncompressed_size': 9423063.333333334,
+                'path_in_schema': 'l_partkey'},
+                {'total_compressed_size': 9405796.666666666,
+                'total_uncompressed_size': 9405346.666666666,
+                'path_in_schema': 'l_suppkey'},
+                ...
+            ]
+        }
+
+        Returns
+        -------
+        dict
+        """
+        idxs = self.sample_statistics()
+        files_to_consider = np.array(self._dataset_info["all_files"])[idxs]
+        stats = [_STATS_CACHE[tokenize(finfo)] for finfo in files_to_consider]
+        return _combine_stats(stats)
+
+    def load_statistics(self, files=None, fragments=None):
+        if files is None:
+            files = self._dataset_info["all_files"]
+        if fragments is None:
+            fragments = self.fragments_unsorted
+        # Collecting code samples is actually a little expensive (~100ms) and
+        # we'd like this thing to be as low overhead as possible
+        with dask.config.set({"distributed.diagnostics.computations.nframes": 0}):
+            token_stats = flatten(
+                dask.compute(_collect_statistics_plan(files, fragments))
+            )
```

Comment:
I moved this from using a sync executor to using the default scheduler. initially, I figured it is cheaper to just run this locally since the collect statistics thing internally will also use a threading executor. However, dispatching to a cluster, if available, makes sense since the cluster is more likely to be closer to the data than the client.

This entire thing adds an overhead of about 300-400ms to the optimize. This is happening in the tune_up step